src/qint.c

```diff
@@ -109,26 +122,45 @@ QINT_API size_t qint_encode4(BufferWriter *bw, uint32_t i1, uint32_t i2, uint32_
   return ret;
 }
 
-#define QINT_DECODE_VALUE(lval, bits, ptr, nused) \
-  do {                                            \
-    switch (bits) {                               \
-      case 2:                                     \
-        lval = *(uint32_t *)ptr & 0xFFFFFF;       \
-        nused = 3;                                \
-        break;                                    \
-      case 0:                                     \
-        lval = *ptr;                              \
-        nused = 1;                                \
-        break;                                    \
-      case 1:                                     \
-        lval = *(uint32_t *)ptr & 0xFFFF;         \
-        nused = 2;                                \
-        break;                                    \
-      default:                                    \
-        lval = *(uint32_t *)ptr;                  \
-        nused = 4;                                \
-        break;                                    \
-    }                                             \
+/* Encode a pair of (32,64) integer */
+QINT_API size_t qint_encode32_64pair(BufferWriter *bw, uint32_t n32, uint64_t n64) {
```

Comment:
I don't mind this trick, but just break the 64 bit number into two 32 bit numbers and encode them with encode3, for code cleanliness. This feels really hacky.